### DISCUSSION TITLE
Limit for how long password is cached?

### DISCUSSION THREAD

**Discussion by lokesh-krishna** [zeapo/Android-Password-Store#1908] [Q&A] [answered]
Recently upgraded to Android 12 and finished setting up Android Store (1.13.5) to find that my password is now cached indefinitely after the first time I entered it. I did pick SSH as my authentication mode over OpenKeychain but that was because picking OpenKeychain wouldn't let me pick my key. The dialog box that popped up didn't display any keys at all. 

I also couldn't find any setting for how long the key is cached. The way things worked in Android 11 was that I had to enter my password every time I opened the app after unlocking my device and that it would remain cached until the screen was locked again. Now I've not had to enter my password a single time since I first input it. Unlocking my phone means access to all my passwords :/

How can I make it so that it behaves the way it used to previously? What other information can I provide regarding my setup to help troubleshoot this or is this intentional behavior?

**Comment by msfjarvis** [comment, marked as answer]
You'll have to configure that within OpenKeychain, since Password Store never receives your key passphrase in the first place. When you're prompted for the passphrase there's a drop-down below the input field that asks how long do you wish for it to be cached.

**Comment by lokesh-krishna** [comment]
Is there a way to get that prompt to show up again?

Edit: Reset OpenKeychain and that made the prompt show up!
